pkg/simple/client/devops/jenkins/pipeline.go

```diff
@@ -661,54 +662,75 @@ func (p *Pipeline) CheckScriptCompile() (*devops.CheckScript, error) {
 }
 
 func (p *Pipeline) CheckCron() (*devops.CheckCronRes, error) {
-
 	var res = new(devops.CheckCronRes)
 
-	Url, err := url.Parse(p.Jenkins.Server + p.Path)
-
+	cronServiceURL, err := url.Parse(p.Jenkins.Server + p.Path)
 	reqJenkins := &http.Request{
 		Method: http.MethodGet,
-		URL:    Url,
+		URL:    cronServiceURL,
 		Header: p.HttpParameters.Header,
 	}
 
 	client := &http.Client{Timeout: 30 * time.Second}
 
+	reqJenkins.SetBasicAuth(p.Jenkins.Requester.BasicAuth.Username, p.Jenkins.Requester.BasicAuth.Password)
 	resp, err := client.Do(reqJenkins)
-
-	if resp != nil && resp.StatusCode != http.StatusOK {
-		resBody, _ := getRespBody(resp)
-		return &devops.CheckCronRes{
-			Result:  "error",
-			Message: string(resBody),
-		}, err
-	}
 	if err != nil {
 		klog.Error(err)
-		return nil, err
+		return interanlErrorMessage(), err
 	}
-	defer resp.Body.Close()
 
-	doc, err := goquery.NewDocumentFromReader(resp.Body)
+	var responseText string
+	if resp != nil {
+		responseData, _ := getRespBody(resp)
+		responseText = string(responseData)
+
+		defer func() {
+			_ = resp.Body.Close()
+		}()
+
+		statusCode := resp.StatusCode
+		if statusCode != http.StatusOK && statusCode != http.StatusBadRequest {
+			// the response body is meaningless for the users, but it's useful for a contributor
+			klog.Errorf("cron service from Jenkins is unavailable, error response: %v, status code: %d", responseText, statusCode)
+			return interanlErrorMessage(), err
+		}
+	}
+	klog.Infof("response text: %s", responseText)
```

Comment:
Always use `klog.Infof()` with an appropriate  level `klog.V(8).Infof()`, or it will flood logs

Reply:
fixed it at https://github.com/kubesphere/kubesphere/pull/3165/commits/730657166c4fbd577d2506a10bbbd9a38a3b9c1a